prow/config/tide.go

```diff
@@ -213,38 +213,81 @@ type TideQuery struct {
 	ReviewApprovedRequired bool `json:"reviewApprovedRequired,omitempty"`
 }
 
-// Query returns the corresponding github search string for the tide query.
-func (tq *TideQuery) Query() string {
-	toks := []string{"is:pr", "state:open", "archived:false"}
+// constructQuery returns a map[org][]orgSpecificQueryParts (org, repo, -repo), remainingQueryString
+func (tq *TideQuery) constructQuery() (map[string][]string, string) {
+	// map org->repo directives (if any)
+	orgScopedIdentifiers := map[string][]string{}
 	for _, o := range tq.Orgs {
-		toks = append(toks, fmt.Sprintf("org:\"%s\"", o))
+		if _, ok := orgScopedIdentifiers[o]; !ok {
+			orgScopedIdentifiers[o] = []string{fmt.Sprintf(`org:"%s"`, o)}
+		}
 	}
 	for _, r := range tq.Repos {
-		toks = append(toks, fmt.Sprintf("repo:\"%s\"", r))
+		if org, _, ok := splitOrgRepoString(r); ok {
+			orgScopedIdentifiers[org] = append(orgScopedIdentifiers[org], fmt.Sprintf("repo:\"%s\"", r))
+		}
 	}
+
 	for _, r := range tq.ExcludedRepos {
-		toks = append(toks, fmt.Sprintf("-repo:\"%s\"", r))
+		if org, _, ok := splitOrgRepoString(r); ok {
+			orgScopedIdentifiers[org] = append(orgScopedIdentifiers[org], fmt.Sprintf("-repo:\"%s\"", r))
+		}
 	}
+
+	queryString := []string{"is:pr", "state:open", "archived:false"}
 	if tq.Author != "" {
-		toks = append(toks, fmt.Sprintf("author:\"%s\"", tq.Author))
+		queryString = append(queryString, fmt.Sprintf("author:\"%s\"", tq.Author))
 	}
 	for _, b := range tq.ExcludedBranches {
-		toks = append(toks, fmt.Sprintf("-base:\"%s\"", b))
+		queryString = append(queryString, fmt.Sprintf("-base:\"%s\"", b))
 	}
 	for _, b := range tq.IncludedBranches {
-		toks = append(toks, fmt.Sprintf("base:\"%s\"", b))
+		queryString = append(queryString, fmt.Sprintf("base:\"%s\"", b))
 	}
 	for _, l := range tq.Labels {
-		toks = append(toks, fmt.Sprintf("label:\"%s\"", l))
+		queryString = append(queryString, fmt.Sprintf("label:\"%s\"", l))
 	}
 	for _, l := range tq.MissingLabels {
-		toks = append(toks, fmt.Sprintf("-label:\"%s\"", l))
+		queryString = append(queryString, fmt.Sprintf("-label:\"%s\"", l))
 	}
 	if tq.Milestone != "" {
-		toks = append(toks, fmt.Sprintf("milestone:\"%s\"", tq.Milestone))
+		queryString = append(queryString, fmt.Sprintf("milestone:\"%s\"", tq.Milestone))
 	}
 	if tq.ReviewApprovedRequired {
-		toks = append(toks, "review:approved")
+		queryString = append(queryString, "review:approved")
+	}
+
+	return orgScopedIdentifiers, strings.Join(queryString, " ")
+}
+
+func splitOrgRepoString(orgRepo string) (string, string, bool) {
+	split := strings.Split(orgRepo, "/")
+	if len(split) != 2 {
+		// Just do it like the github search itself and ignore invalid orgRepo identifiers
+		logrus.WithField("OrgRepoString", orgRepo).Warn("OrgRepo string is invalid, will be ignored")
```

Comment:
Isn't this going to be excessively spammy if something is misconfigured?

Reply:
Yeah, removed if. Ignoring something like that is effectively what already happens today, as github just won't find anything in that case